sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2965,4 +2965,32 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-16602 Nvl/Coalesce") {
```

Comment:
We can just test it in expression unit test by calling replaceForTypeCoercion, can't we?